drivers: pcie: endpoint: warning generated from pcie_ep_iproc

**Describe the bug**

With the recent change that went into `pcie_ep_iproc.c` in #57305, there is a remaining warning generated.

It was not caught with the initial change that accounted for the optional DMA controllers, and we do not yet have a continuous build for this driver.

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using? `bcm958402m2_m7`
 - What have you tried to diagnose or workaround this issue? pulled the most recent `v2.7-branch` internally, ran a build.
 - Is this a regression? Not really, it's just a warning, so mainly an annoyance.
 - ...

**To Reproduce**
Same steps as in https://github.com/zephyrproject-rtos/zephyr/issues/57210

**Expected behavior**
No warnings.

**Impact**
Annoyance

**Logs and console output**
```shell
/home/cfriedt/zephyrproject/zephyr/drivers/pcie/endpoint/pcie_ep_iproc.c:176:12: warning: 'iproc_pcie_pl330_dma_xfer' defined but not used [-Wunused-function]
  176 | static int iproc_pcie_pl330_dma_xfer(const struct device *dev,
```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows): Linux
 - Toolchain (e.g Zephyr SDK, ...): SDK v0.16.0
 - Commit SHA or Version used: 1bcffb2a45cab74e1f13e1447ab912012f4e878e

**Additional context**

--- a/drivers/pcie/endpoint/pcie_ep_iproc.c
+++ b/drivers/pcie/endpoint/pcie_ep_iproc.c
@@ -175,6 +175,7 @@ static int iproc_pcie_register_reset_cb(const struct device *dev,
 	return 0;
 }
 
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(dmas)
 static int iproc_pcie_pl330_dma_xfer(const struct device *dev,
 				     uint64_t mapped_addr,
 				     uintptr_t local_addr, uint32_t size,
@@ -222,6 +223,7 @@ static int iproc_pcie_pl330_dma_xfer(const struct device *dev,
 out:
 	return ret;
 }
+#endif
 
 #if DT_INST_IRQ_HAS_NAME(0, perst)
 static void iproc_pcie_perst(const struct device *dev)

